README.md

```diff
@@ -84,17 +117,94 @@ logout (storage?: Storage): Promise<void>
 
 Clears the active user session.
 
-Note: this is an unsupported use case in WebID-TLS.  Once your browser provides
-its client cert to a web server, there's no going back!  So for WebID-TLS, the
-only thing this will do is clear the session from the store.
+WARNING: this is an unsupported use case in WebID-TLS.  Once your browser
+provides its client cert to a web server, there's no going back!  So for
+WebID-TLS, the only thing this will do is clear the session from the store.
 
 ### `fetch`
 
 Fetches a resource from the web.  Same API as
-[fetch](https://fetch.spec.whatwg.org/), but retries with credentials when
-it encounters a `401` with a `WWW-Authenticate` header which matches a
-recognized authenticate scheme.
+[fetch](https://fetch.spec.whatwg.org/), but retries with credentials when it
+encounters a `401` with a `WWW-Authenticate` header which matches a recognized
+authenticate scheme.
 
 ```
 fetch: (url: RequestInfo, options?: Object) => Promise<Response>
 ```
+
+## Developing
+
+### Prerequisites
+
+This library assumes you have [node](https://nodejs.org/en/) >= v7.10.1  and
+[yarn](https://yarnpkg.com/) 0.24.6 installed.  It may work with earlier
+versions, but that hasn't been tested thus far.
+
+### Setting up the development environment
+
+```sh
+$ git clone https://github.com/solid/solid-auth-client.git
+$ cd solid-auth-client
+$ yarn
+$ yarn build # build the library and UMD bundle
+$ yarn test # run the code formatter, linter, and test suite
+$ yarn test:dev # just run the tests in watch mode
+```
+
+### Building the demo app
+
+```sh
+$ cp .env.demo.example .env.demo && $EDITOR .env.demo # configure the demo app
+$ yarn start:demo
+```
+
+#### Configuration
+
+The demo app is configurable via the `.env.demo` file.  The important fields are:
+
+- IDP_SELECT_URI: URI for the popup-based IDP select app.  When testing
+  locally, this will be something like 'http://localhost:XXXX/idp-select.html'
+- CALLBACK_URI: URI for the popup-based callback app.  When testing locally,
+  this will be something like 'http://localhost:XXXX/idp-callback.html'
+
+### Building the popup app
+
+```sh
+$ cp .env.popup.example .env.popup && $EDITOR .env.popup # configure the popup app
+$ yarn start:popup
```

Comment:
I was confused here that this didn't just give me the HTML files I need, but instead starts an app. (I mean, the "start" part is named well, but it's just not what I expect.)

How can I generate the files?

Reply:
It's also weird that both the popup and demo are started on 8080.